integrations/webhook/src/index.ts

```diff
@@ -0,0 +1,153 @@
+import { createIntegration, Logger } from '@gitbook/runtime';
+import { Event } from '@gitbook/api';
+
+import { configComponent } from './config';
+import {
+    MAX_RETRIES,
+    REQUEST_TIMEOUT,
+    WebhookRuntimeContext,
+    retryWithDelay,
+    EventType,
+    EVENT_TYPES,
+} from './common';
+
+const logger = Logger('webhook');
+
+/**
+ * Common webhook delivery handler for all event types
+ */
+const handleWebhookEvent = async (event: Event, context: WebhookRuntimeContext) => {
+    const { environment } = context;
+    const spaceInstallation = environment.spaceInstallation;
+
+    if (!spaceInstallation) {
+        logger.debug('No space installation found');
+        return;
+    }
+
+    const config = spaceInstallation.configuration;
+
+    // Check if this event type is supported and enabled
+    if (!EVENT_TYPES.includes(event.type as EventType) || !config[event.type as EventType]) {
+        logger.debug(`Event ${event.type} is not enabled`);
+        return;
+    }
+
+    // Prepare webhook payload
+    const jsonPayload = JSON.stringify(event);
+
+    // Add HMAC signature for webhook verification
+    const encoder = new TextEncoder();
+    const key = await crypto.subtle.importKey(
+        'raw',
+        encoder.encode(config.secret),
+        { name: 'HMAC', hash: 'SHA-256' },
+        false,
+        ['sign'],
+    );
+    const signatureBuffer = await crypto.subtle.sign('HMAC', key, encoder.encode(jsonPayload));
```

Comment:
Would it be worth generating the hash based also on a timestamp, at which we send the event, to prevent against replay attacks? We could use similar format as Stripe uses e.g: `<timestamp>.<JSONPayload>` for the payload that we hash and include the timestamp in the signature header.

I know that customers should make sure that their webhook handlers are idempotent to prevent replay attacks but generating timestamp based signature would enforce protection if they don't ensure idempotency.

Reply:
That's not a bad idea, I'll give it a shot 👌🏼